service-base/src/main/java/org/eclipse/hono/service/http/DefaultFailureHandler.java

```diff
@@ -75,6 +76,32 @@ public void handle(final RoutingContext ctx) {
         }
     }
 
+    private String extractMessage(final HttpServerResponse response) {
+        if ( response == null ) {
+            return MSG_NO_ERROR;
+        }
+
+        final String message = response.getStatusMessage();
+        if ( message == null ) {
+            return Integer.toString(response.getStatusCode());
+        }
+
+        return message;
+    }
+
+    private String extractMessage(final Throwable cause) {
+        if (cause == null) {
+            return MSG_NO_ERROR;
+        }
+
+        final String message = cause.getMessage();
+        if (message == null) {
+            return cause.getClass().getName();
```

Comment:
I would rather not expose the exception class name here in order not to encourage people to actually rely on the class names...

Reply:
I had this at first ... the problem with that seems to me that you would have no clue at all what went wrong. IMHO known the class name of an Exception doesn't really help you get any insight at all.

Reply:
Which view are you taking? A Hono developer? A user of a hosted Hono instance?

Reply:
A *user* shouldn't see the internal class names. A *developer* can take a look at the (server side) logs.